src/mixed-line-bar-chart/domain.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { InternalChartSeries, MixedLineBarChartProps, ScaleType } from './interfaces';
+import { isDataSeries, isXThreshold, isYThreshold, matchesX } from './utils';
+
+export function computeDomainX<T>(series: readonly InternalChartSeries<T>[], xScaleType: ScaleType) {
+  if (xScaleType === 'categorical') {
+    return series.reduce((acc, s) => {
+      if (isDataSeries(s.series)) {
+        s.series.data.forEach(({ x }) => {
+          if (acc.indexOf(x) === -1) {
+            acc.push(x);
+          }
+        });
+      }
+      if (isXThreshold(s.series)) {
+        if (acc.indexOf(s.series.x) === -1) {
+          acc.push(s.series.x);
+        }
+      }
+      return acc;
+    }, [] as T[]);
+  }
+
+  return series.reduce((acc, curr) => {
+    // Y-thresholds don't have X value.
+    if (isYThreshold(curr.series)) {
+      return acc;
+    }
+
+    // Compare x-threshold X with current min, max.
+    if (isXThreshold(curr.series)) {
+      const [min, max] = acc;
+      const newMin = min === undefined || curr.series.x < min ? curr.series.x : min;
+      const newMax = max === undefined || max < curr.series.x ? curr.series.x : max;
+      return [newMin, newMax] as T[];
+    }
+
+    // Compare all series X values with current min, max.
+    if (isDataSeries(curr.series)) {
+      return curr.series.data.reduce(([min, max], { x }) => {
+        const newMin = min === undefined || x < min ? x : min;
+        const newMax = max === undefined || max < x ? x : max;
+        return [newMin, newMax] as T[];
+      }, acc);
+    }
+
+    return acc;
+  }, [] as T[]);
+}
+
+function find<Q>(arr: readonly Q[], func: (el: Q) => boolean) {
+  for (let i = 0; i < arr.length; i++) {
+    const found = func(arr[i]);
+    if (found) {
+      return arr[i];
+    }
+  }
+  return null;
+}
+
+export function computeDomainY<T>(
+  series: readonly InternalChartSeries<T>[],
+  scaleType: 'linear' | 'log',
+  stackedBars: boolean
+) {
+  let _series = series;
+
+  // For stacked bars, we need to accumulate all the bar series into a positive and a negative series
+  if (stackedBars) {
+    const { positiveData, negativeData } = series.reduce(
+      (acc, curr) => {
+        if (curr.series.type === 'bar') {
+          curr.series.data.forEach(({ x, y }) => {
+            const data = y < 0 ? acc.negativeData : acc.positiveData;
+            const stackedDatum = find(data, el => matchesX(el.x, x));
+            if (stackedDatum) {
+              stackedDatum.y += y;
+            } else {
+              data.push({ x, y });
+            }
+            return acc;
```

Comment:
This is misplaced. Really correct `return` is below.

Reply:
that's the original logic, see: https://github.com/cloudscape-design/components/blob/main/src/mixed-line-bar-chart/utils.ts#L70

Reply:
Still doesn't make sense to `return` inside a `forEach`.